examples/SampleApp/src/hooks/usePaginatedPinnedMessages.ts

```diff
@@ -0,0 +1,105 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+
+import { AppContext } from '../context/AppContext';
+
+import type { Channel, MessageResponse } from 'stream-chat';
+
+import type {
+  LocalAttachmentType,
+  LocalChannelType,
+  LocalCommandType,
+  LocalEventType,
+  LocalMessageType,
+  LocalReactionType,
+  LocalUserType,
+} from '../types';
+
+export const usePaginatedPinnedMessages = (
+  channel: Channel<
+    LocalAttachmentType,
+    LocalChannelType,
+    LocalCommandType,
+    LocalEventType,
+    LocalMessageType,
+    LocalReactionType,
+    LocalUserType
+  >,
+) => {
+  const { chatClient } = useContext(AppContext);
+  const offset = useRef(0);
+  const hasMoreResults = useRef(true);
+  const queryInProgress = useRef(false);
+  const [loading, setLoading] = useState(true);
+  const [messages, setMessages] = useState<
+    MessageResponse<
+      LocalAttachmentType,
+      LocalChannelType,
+      LocalCommandType,
+      LocalMessageType,
+      LocalReactionType,
+      LocalUserType
+    >[]
+  >([]);
+
+  const fetchPinnedMessages = async () => {
+    if (queryInProgress.current) {
+      return;
+    }
+
+    setLoading(true);
+
+    try {
+      queryInProgress.current = true;
+
+      offset.current = offset.current + messages.length;
+
+      if (!hasMoreResults.current) {
+        queryInProgress.current = false;
+        return;
+      }
+
+      // TODO: Use this when support for attachment_type is ready.
+      const res = await chatClient?.search(
+        {
+          cid: { $in: [channel.cid] },
+        },
+        { pinned: true },
+        {
+          limit: 10,
```

Comment:
Bit of a nitpick: Where does the limit of 10 here come from? Should it be possible to change it? Don't know if it's of any value, but I'm vuessing this is related to the check of whether there are fewer than 10 new messages a bit lower down, in which case it might be worth setting it as a const so it's easier to change both and see that the two values are related (if I'm right?)

Reply:
This is for pagination in the Pinned Messages Screen. This will load 10 instances of data at one time This was the case in the Attachment hook too so I used it here. @vishalnarkhede is it fine?

Reply:
You mean I should replace:
```tsx
if (!hasMoreResults.current) {
  queryInProgress.current = false;
  return;
}
```
with the one mentioned above?